test/t/conftest.py

```diff
@@ -512,7 +513,33 @@ def _save_cwd(self):
             self.cwd_changed = True
             self._copy_variable("PWD", "%s_OLDPWD" % self.prefix)
 
+    def _check_set(self, name: str):
+        if self.saved_set[name] != 1:
```

Comment:
Let's see if we can find more meaningful values for this than magic 1 and 2 ints. An enum?

(saved_shopt and saved_variables on the other hand look like they would be better done as `[str, bool]`, but that's out of scope here.)

Reply:
I have introduced an enum in 34fdc3435ad7cae8fffe993db6e73135bf506b12